scanpy_scripts/lib/_filter.py

```diff
@@ -50,41 +50,41 @@ def filter_anndata(
         return 0
 
     conditions, qc_vars, pct_top = _get_filter_conditions(
-        adata, attributes, param, category, subset)
+        attributes, param, category, subset)
 
-    sc.pp.filter_cells(adata, min_genes=0)
-    sc.pp.filter_cells(adata, min_counts=0)
-    sc.pp.filter_genes(adata, min_cells=0)
-    sc.pp.filter_genes(adata, min_counts=0)
+    if 'n_genes' not in adata.obs.columns:
+        sc.pp.filter_cells(adata, min_genes=0)
+    if 'n_counts' not in adata.obs.columns:
+        sc.pp.filter_cells(adata, min_counts=0)
+    if 'n_cells' not in adata.var.columns:
+        sc.pp.filter_genes(adata, min_cells=0)
+    if 'n_counts' not in adata.var.columns:
+        sc.pp.filter_genes(adata, min_counts=0)
 
     if qc_vars or pct_top:
         if not pct_top:
-            pct_top = [50]
+            pct_top = [1]
```

Comment:
Why this change?

Reply:
> Why the switch from cell -> c and gene -> g? Wouldn't it be more descriptive to leave the full words?

It's less typing on the command line which cli users will care (I know workflow users won't :-) ), and the prefix can be hidden by the galaxy layer so change is transparent to galaxy users.

Reply:
> Why this change?

Good catch. Was thinking of removing the whole default value thing, as it was a work around for an old bug in scanpy, but decided to leave it but incorrectly left 1 there.